internal/daemonset/daemonset.go

```diff
@@ -54,7 +54,7 @@ func (dc *daemonSetGenerator) GarbageCollect(ctx context.Context, existingDS map
 	deleted := make([]string, 0)
 
 	for kernelVersion, ds := range existingDS {
-		if !validKernels.Has(kernelVersion) {
+		if !dc.isDevicePluginDaemonSet(ds) && !validKernels.Has(kernelVersion) {
```

Comment:
I think we could instead make `existingDS` contain only driver DaemonSets. WDYT?

Reply:
it used to previously, but i changed it to contain also device plugin, since it allows us to use this info in the status updater and metrics, without additional queries

Reply:
We agree to refactor if this ever leads to many ifs in the codebase.